### DISCUSSION TITLE
troubles experimenting with issuer-api

### DISCUSSION THREAD

**Discussion by janducaju** [walt-id/.github#77] [Help]
Hi,
I’m experimenting with walt.id issuer-api. I deployed conform https://docs.oss.walt.id/issuer/api/setup/open-source 
As our testing docker environment is running IPv6 only so I started the container like
docker run --rm --name issuer-api  waltid/issuer-api   --webHost=::   --webPort=7002   --baseUrl=http://icts-t-dev-u0001439-issuer-api.service.qsvcd:7002
where icts-t-dev-u0001439-issuer-api.service.qsvcd is the address that will be running the docker image
instead of (like documented)
docker run -p 7002:7002 waltid/issuer-api \
  --webHost=0.0.0.0 \
  --webPort=7002 \
  --baseUrl=http://localhost:7002
When I try to issue an UniversityDegreeCredential:
![attachment1](https://github.com/walt-id/.github/assets/14836916/023ad690-5274-4c13-958c-f27b03e57e8c)
When I paste the response body in the wallet I get:
![attachment2](https://github.com/walt-id/.github/assets/14836916/46186e6f-7981-490f-9346-907070381604)
When I click on Accept I get:
 wallet.walt.id says: error occurred while trying to receive credential: [object: Object]
![attachment3](https://github.com/walt-id/.github/assets/14836916/0b6e9b08-e260-488d-9880-539d2d81cc66)

Do I need to run the docker image differently (or am I missing something else)
adTHANKSvance,
Jan

**Comment by ahmedsmiai** [comment]
hey @janducaju, to resolve this, you can also host both the Web Wallet and the Wallet Backend on your testing docker environment, allowing them to communicate with the issuer service, However, you'll need to ensure that the issuer service is configured to provide its public address to the wallet during the issuance request. Additionally, it's likely that the public address needs to be served over HTTPS to avoid rejection.

**Reply by janducaju** [reply]
thx, ahmedsmiai
Is there documentation to let it work with the wallet https://wallet.walt.id/ without hosting the Web Wallet and Wallet Backend in my docker environment?
I will try to deploy the issuer-api container with certificate and a public address.
The same for the verifier-api as at the moment when I have a UniversityDegreeCredential in my wallet via https://issuer.portal.walt.id/swagger/index.html I get "Undefined" when verifying it
![image](https://github.com/walt-id/.github/assets/14836916/4f0cd9a5-ad5e-4988-9a30-e5a59abbb647)
![Picture1](https://github.com/walt-id/.github/assets/14836916/21a3c3f2-bbdb-45df-bde9-32e7603dacd6)

**Comment by philpotisk** [comment]
Hi @janducaju , I think the easiest way to run the entire stack is using Docker Compose as described here: https://github.com/walt-id/waltid-identity/tree/main/docker-compose

In production we recommend using Kubernetes. We could also provide you with the Kubernetes configuration if this would be helpful for you.

**Reply by janducaju** [reply]
Hi philpotisk

Thanks but I would like to test between locally deployed docker containers for verify-api and issuer-api and on the other hand the PWA wallet on the smartphone https://wallet.walt.id/ or https://wallet.walt-test.cloud/
By the way which of the both above is best to test https://wallet.walt-test.cloud/ or https://wallet.walt.id/

**Reply by alegomes** [reply]
> By the way which of the both above is best to test https://wallet.walt-test.cloud/ or https://wallet.walt.id/

https://wallet.walt.id

**Comment by alegomes** [comment]
This forum is no longer being monitored. Please, if you still need our help, submit your question as a [Community Stack issue](https://github.com/walt-id/waltid-identity/issues).

All the best
